grpc-server/grpc_server.go

```diff
@@ -7,148 +7,88 @@ import (
 	"net"
 	"os"
 	"path/filepath"
-	"strings"
-	"sync"
 	"time"
 
 	grpc_middleware "github.com/grpc-ecosystem/go-grpc-middleware"
 	grpc_prometheus "github.com/grpc-ecosystem/go-grpc-prometheus"
-	"github.com/packethost/pkg/log"
 	"github.com/pkg/errors"
-	"github.com/tinkerbell/tink/db"
-	"github.com/tinkerbell/tink/protos/hardware"
-	"github.com/tinkerbell/tink/protos/template"
-	"github.com/tinkerbell/tink/protos/workflow"
 	"go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc"
 	"google.golang.org/grpc"
-	"google.golang.org/grpc/credentials"
 	"google.golang.org/grpc/reflection"
 )
 
-// Server is the gRPC server for tinkerbell.
-type server struct {
-	cert []byte
-	modT time.Time
+// GetCerts returns a TLS certificate, PEM bytes, and file modification time for a
+// given path. An error is returned for any failure.
+//
+//
+// The public key is expected to be named "bundle.pem" and the private key
+// "server.pem".
+func GetCerts(certsDir string) (*tls.Certificate, []byte, *time.Time, error) {
+	certFile, err := os.Open(filepath.Join(certsDir, "bundle.pem"))
+	if err != nil {
+		return nil, nil, nil, errors.Wrap(err, "failed to open TLS cert")
+	}
 
-	db   db.Database
-	quit <-chan struct{}
+	stat, err := certFile.Stat()
+	if err != nil {
+		return nil, nil, nil, errors.Wrap(err, "failed to stat TLS cert")
+	}
+	modT := stat.ModTime()
+	certPEM, err := ioutil.ReadAll(certFile)
+	if err != nil {
+		return nil, nil, nil, errors.Wrap(err, "failed to read TLS cert")
+	}
+	err = certFile.Close()
+	if err != nil {
+		return nil, nil, nil, errors.Wrap(err, "failed to close TLS cert")
+	}
 
-	dbLock  sync.RWMutex
-	dbReady bool
+	keyPEM, err := ioutil.ReadFile(filepath.Join(certsDir, "server-key.pem"))
+	if err != nil {
+		return nil, nil, nil, errors.Wrap(err, "failed to read TLS key")
+	}
 
-	watchLock sync.RWMutex
-	watch     map[string]chan string
+	cert, err := tls.X509KeyPair(certPEM, keyPEM)
+	if err != nil {
+		return nil, nil, nil, errors.Wrap(err, "failed to parse TLS file content")
+	}
 
-	logger log.Logger
+	return &cert, certPEM, &modT, nil
 }
 
-type ConfigGRPCServer struct {
```

Comment:
Removing the `ConfigGRPCServer` type would be a breaking change. 
Can we call this out better along with an example of how to adapt?

Reply:
Sure

Reply:
Added example in PR description